### PR TITLE
proto: Add language server support

### LINKED ISSUE #18762: Add language server support to newly created proto extension
### Check for existing issues

- [X] Completed

### Describe the feature

As discussed with @maxdeviant in zed-industries/extensions#1479 I would like to add basic language server support to the newly created (thanks again!) proto extension.

The language server in question is [this one](/asorda/protobuf-language-server).

### PR DESCRIPTION
**Pull request by prskr**
Closes #18762

Release Notes:

- N/A

### CODE REVIEW

File: extensions/proto/src/proto.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+struct ProtobufExtension {}
+
+impl ProtobufExtension {
+    fn language_server_binary(
+        &self,
+        _language_server_id: &zed::LanguageServerId,
+        worktree: &zed::Worktree,
+    ) -> Result<ProtobufLspBinary> {
+        let binary_settings = LspSettings::for_worktree("omnisharp", worktree)
```
Comment (notpeter) [inline extensions/proto/src/proto.rs:18]: s/omnisharp/protobuf-language-server/

Comment (prskr) [reply extensions/proto/src/proto.rs:18]: that is quite embarrassing :D thanks!

Comment (prskr) [reply extensions/proto/src/proto.rs]: fixed 😊

**Review by maxdeviant — APPROVED** [review]
Thank you!

**Comment by alexkirsz** [comment]
Hello! If I'm not mistaken, this first implementation does not allow for passing settings to the LSP?

**Comment by prskr** [comment]
TBH I assumed that initialization options would be independent as this is not depending on a specific LS or are you referring to additional CLI switches?

**Pull request merged**